### PR TITLE
Provide real authorization on requests

### PR DESCRIPTION
**Pull request by tabfugnic**

### CODE REVIEW

File: app/models/icims/client.rb [hunk 1]
```diff
@@ -43,16 +43,27 @@ def candidates
...
+          search_params.to_json,
+          authorized_params,
         )
       end
 
       def authorized_params
-        { key: connection.key }
+        {
+          authorization: connection.key,
+          content_type: "application/json"
```
Comment (hrs) [inline app/models/icims/client.rb]: Trailing comma!

File: app/models/icims/client.rb [hunk 2]
```diff
@@ -70,12 +81,13 @@ def search_params
           filters: [
             {
               name: "person.employeeinfo.hiredate",
-              operator: "=",
+              operator: "<",
```
Comment (hrs) [inline app/models/icims/client.rb]: I don't think I understand this any better. Less than what?

File: spec/features/user_imports_icims_people_spec.rb [hunk 3]
```diff
@@ -9,6 +12,15 @@
   end
 
   scenario "successfully imports candidates" do
+    user = create(:user)
+    create(
+      :icims_connection,
+      :connected,
+      user: user,
+      username: USERNAME,
+      password: PASSWORD
```
Comment (hrs) [inline spec/features/user_imports_icims_people_spec.rb]: Trailing comma!

File: spec/features/user_imports_icims_people_spec.rb [hunk 4]
```diff
@@ -1,6 +1,9 @@
 require "rails_helper"
 
 feature "User imports icims people" do
+  USERNAME = "crashoverride"
```
Comment (hrs) [inline spec/features/user_imports_icims_people_spec.rb]: I might make these methods instead.

**Comment by hrs** [comment]
Once these comments are resolved I think we'll be ready to merge!

**Comment by tabfugnic** [comment]
Much too dated to consider using anymore

**Pull request closed without merging**